src/arch/forte_architecture.h

```diff
@@ -15,20 +15,80 @@
 #ifndef SRC_ARCH_FORTE_ARCHITECTURE_H_
 #define SRC_ARCH_FORTE_ARCHITECTURE_H_
 
-class CForteArchitecture{
+#include "forte_specific_architecture.h"
+
+#include "forteinit.h"
+#include "startuphook.h"
+
+template<typename SpecificArchitectureT>
```

Comment:
It seems overkill to me to use a template parameter on top of conditional compilation. What is the benefit instead of just using `CForteSpecificArchitecture` directly?

Reply:
My main motivation was usability. When you are using the architecture inside forte, the specifics of the architecture should not be visible to the user. By having a "general" architecture, we can define a proper interface with a clear documentation of what it does. 

Specific architectures might add variables or other private methods which might distract the user. 

Also, when creating a new architecture, you should only look at the "general" architecture and nothing else.

Reply:
I guess I worded that poorly. I am not against splitting this into a general and specific part, far from it. I am merely wondering why we need a template parameter instead of just using `CForteSpecificArchitecture` **inside** `CForteGeneralArchitecture`.

Reply:
I see. I completely missed that. You're right. It's updated now